base/rtlib.jl

```diff
@@ -0,0 +1,194 @@
+# This file is a part of Julia. License is MIT: http://julialang.org/license
+"""
+    RTLIB
+
+Implements the runtime library for Julia. The implementations are based on
+llvm's compiler-rt. This implementations follows the compiler-rt naming convention
+and registers the pure Julia implementation as `extern_c` so that LLVM can find them.
+
+As a secondary interface `RTLIB.convert(::Type{T}, x)` is provided.
+"""
+module RTLIB
+
+register(f::Function, rtype::ANY, argt::ANY, name::String) =
+    ccall(:jl_extern_c, Void, (Any, Any, Any, Cstring),
+          f, rtype, argt, name)
+
+include("rtlib/fp_util.jl")
```

Comment:
This might be a bit tricky because I think relative include only becomes available a bit later. https://github.com/JuliaLang/julia/blob/2618d7dd316cf85539a703da40cfa81623ec9792/base/sysimg.jl#L228
I will see if I can move that to an earlier point.